spec/index.bs

```diff
@@ -3202,49 +3215,52 @@ and {{GPUCommandEncoder/copyTextureToBuffer()}}.
+Issue: define this as an algorithm with (texture, mipmapLevel) parameters and use the call syntax instead of referring to the definition by labbl.
 
+<div algorithm class=validusage>
+    <dfn abstract-op>validating linear texture data</dfn>(layout, byteSize, format, copyExtent)
+    **Arguments:**
+        - {{GPUTextureDataLayout}} |layout| of the linear texture data
+        - `GPUSize64` |byteSize| - total size of the linear data, in bytes
+        - {{GPUTextureFormat}} |format| of the texture
+        - {{GPUExtent3D}} |copyExtent| - extent of the texture to copy
+
+    Let:
+        - |blockWidth| be the [=texel block width=] of |format|.
+        - |blockHeight| be the [=texel block height=] of |format|.
+        - |blockSize| be the [=texel block size=] of |format|.
+        - |bytesInACompleteRow| be |blockSize| &times; |copyExtent|.[=Extent3D/width=] &div; |blockWidth|.
+        - |bytesInACompleteImage| be |bytesInACompleteRow| &times; |copyExtent|.[=Extent3D/height=] &div; |blockHeight|.
+        - |requiredBytesInCopy| be calculated with the following algorithm assuming all the parameters are [=valid=]:
+            ```
+            if (copyExtent.width == 0 || copyExtent.height == 0 || copyExtent.depth == 0) {
+                requiredBytesInCopy = 0;
+            } else {
+                GPUSize64 texelBlockRowsPerImage = layout.rowsPerImage / blockHeight;
+                GPUSize64 bytesPerImage = layout.bytesPerRow * texelBlockRowsPerImage;
+                GPUSize64 bytesInLastSlice =
+                    layout.bytesPerRow * (copyExtent.height / blockHeight - 1) + (copyExtent.width / blockWidth * blockSize);
+                requiredBytesInCopy = bytesPerImage * (copyExtent.depth - 1) + bytesInLastSlice;
+            }
+            ```
+
+    The following validation rules apply:
+
+    For the copy being in-bounds:
+    - If |layout|.{{GPUTextureDataLayout/rowsPerImage}} is not 0, it must be greater than or equal to
+        |copyExtent|.[=Extent3D/height=].
+    - (|layout|.{{GPUTextureDataLayout/offset}} + |requiredBytesInCopy|) must not overflow a {{GPUSize64}}.
+    - (|layout|.{{GPUTextureDataLayout/offset}} + |requiredBytesInCopy|) must be smaller than or equal to |byteSize|.
+
+    For the texel block alignments:
+    - |layout|.{{GPUTextureDataLayout/rowsPerImage}} must be a multiple of |blockHeight|.
+    - |layout|.{{GPUTextureDataLayout/offset}} must be a multiple of |blockSize|.
+
+    For other members in |layout|:
+    - If |copyExtent|.[=Extent3D/height=] is greater than 1:
+      - |layout|.{{GPUTextureDataLayout/bytesPerRow}} must be greater than or equal to the number of |bytesInACompleteRow|.
+    - If |copyExtent|.[=Extent3D/depth=] is greater than 1:
+      - |layout|.{{GPUTextureDataLayout/rowsPerImage}} must be greater than or equal to the number of |bytesInACompleteImage|.
```

Comment:
This compares "rows" to `bytesInACompleteImage`; does `rowsPerImage` need to be converted into bytes first?

Reply:
looks like it was a bug in the current spec that I just copied over to a new location. Fixing here anyway